huggingface_sb3/push_to_hub.py

```diff
@@ -7,30 +7,309 @@
+  ## Evaluation Results
+  """
+
+    model_card += f"""
+  mean_reward={mean_reward:.2f} +/- {std_reward}
+  """
+
+    model_card += """
+  ## Usage (with Stable-baselines3)
+  TODO: Add your code
+  """
+
+    return model_card
+
+
+def _create_model_card(repo_dir: Path, generated_model_card):
+    """Creates a model card for the repository.
+    TODO: Add metrics to model-index
+    TODO: Use information from common model cards
+    :param repo_dir: repository directory
+    :param generated_model_card: model card generated by _generate_model_card() method
     """
     readme_path = repo_dir / "README.md"
     readme = ""
     if readme_path.exists():
-      with readme_path.open("r", encoding="utf8") as f:
-          readme = f.read()
+        with readme_path.open("r", encoding="utf8") as f:
+            readme = f.read()
     else:
-      readme = README_TEMPLATE
+        readme = generated_model_card
     with readme_path.open("w", encoding="utf-8") as f:
         f.write(readme)
 
+
+def package_to_hub(model,
+                   model_name: str,
+                   model_architecture: str,
+                   env_id: str,
+                   eval_env,
+                   repo_id: str,
+                   commit_message: str,
+                   is_deterministic=True,
+                   n_eval_episodes=10,
+                   use_auth_token=True,
+                   local_repo_path="hub",
+                   video_length=1000,
+                   ):
+    """
+      Evaluate, Generate a video and Upload a model to Hugging Face Hub.
+      This method does the complete pipeline:
+      - It evaluates the model
+      - It generates the model card
+      - It generates a replay video of the agent
+      - It pushes everything to the hub
+      This is a work in progress function, if it does not work, use push_to_hub method
+      :param model: trained model
+      :param model_name: name of the model zip file
+      :param model_architecture: name of the architecture of your model (DQN, PPO, A2C, SAC...)
+      :param env_id: name of the environment
+      :param eval_env: environment used to evaluate the agent
+      :param repo_id: repo_id: id of the model repository from the Hugging Face Hub
+      :param commit_message: commit message
+      :param is_deterministic: use deterministic or stochastic actions (by default: True)
+      :param n_eval_episodes: number of evaluation episodes (by default: 10)
+      :param use_auth_token
+      :param local_repo_path: local repository path
+      :param video_length:
+      :param video_length: length of the video (in timesteps)
+      """
+    print(
+        "This function will save, evaluate, generate a video of your agent, create a model card and push everything to the hub. It might take up to 1min. \n This is a work in progress: If you encounter a bug, please send a message to thomas.simonini@huggingface.co and use push_to_hub instead.")
+    huggingface_token = HfFolder.get_token()
```

Comment:
I don't think you need to get the token. All the functions you use get the token under the hood. Passing the token is useful if you plan to make unit tests

Reply:
I was checking on other implementations but are you sure we can remove it?

Reply:
`create_repo` accepts `token=None`, which is totally fine since when that happens it will just retrieve the token saved in the system with `HfFolder.get_token()`, which you're trying to do here.

I think what you likely want is something such as https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/fastai_utils.py#L352

So instead of `use_auth_token`, you would have an optional `token` param and pass it to `create_repo`. This is useful for colabs in which users are passing it, or for tests.

Reply:
:heavy_check_mark: